.github/CONTRIBUTING.md

```diff
@@ -34,151 +30,191 @@ ensure that the community is free to use your contributions.
+other applications and improve your experience.
 
-As we are dealing with languages that build to native binaries, we will
-need to have installed any of the `supported CMake build tools <https://cmake.org/cmake/help/v3.5/manual/cmake-generators.7.html>`_.
+The simplest way to use environments is by using **Anaconda** in a terminal
+window
 
-We do support most of the common available toolchains like: gcc, clang, Visual Studio.
-The only required requisite is that the toolchain needs to support C++14.
+    conda create -y -n QiskitDevEnv python=3
+    source activate QiskitDevEnv
 
-**Mac**
+Clone the Qiskit Aer repo via **git**.
 
-On Mac we have various options depending on the compiler we want to use.
-If we want to use Apple's Clang compiler, we need to install an extra library for
-supporting OpenMP: libomp. The CMake build system will warn you otherwise.
-To install it manually:
-you can type:
+    git clone https://github.com/Qiskit/qiskit-aer
 
-```
-$ brew install libomp
-```
+Most of the required dependencies can be installed via **pip**, using the
+`requirements-dev.txt` file, e.g.:
 
-We do recommend installing OpenBLAS, which is our default choice:
+    cd qiskit-aer
+    pip install -r requirements-dev.txt
 
-```
-$ brew install openblas
-```
 
-CMake build system will search for other BLAS implementation alternatives if
-OpenBLAS is not installed in the system.
+### Linux
 
-You further need to have Command Line Tools installed on MacOS:
+Qiskit is supported on Ubuntu >= 16.04. To get most of the necessary compilers
+and libraries, install the `build-essential` package by running
 
-```
-$ xcode-select --install
-```
+    sudo apt install build-essential
 
+Although the **BLAS** and **LAPACK** library implementations included in the
+`build-essential` package are sufficient to build all of the Aer simulators, we
+recommend using **OpenBLAS**, which you can install by running
 
-**Linux (Ubuntu >= 16.04)**
+    sudo apt install libopenblas-dev
 
-Most of the major distributions come with a BLAS and LAPACK library implementation,
-and this is enough to build all the simulators, but we do recommend using OpenBLAS
-here as well, so in order to install it you have to type:
+There are two ways of building Aer simulators, depending on your goal:
 
-```
-$ sudo apt install libopenblas-dev
-```
+1. Build a Terra compatible add-on;
+2. Build a standalone executable.
+
+**Terra Add-on**
+
+For the former, we just need to call the `setup.py` script:
+
+    qiskit-aer$ python ./setup.py bdist_wheel
+
+We are using **scikit-build** as a substitute for **setuptools**. This is
+basically the glue between **setuptools** and **CMake**, so there are various
+options to pass variables to **CMake**, and the underlying build system
+(depending on your platform). The way to pass variables is:
+
+    qiskit-aer$ python ./setup.py bdist_wheel -- -DCMAKE_VARIABLE=Values -- -
+    Makefile_Flag
```

Comment:
This `Makefile_Flag` should be right after the `-` of the above line:
```
qiskit-aer$ python ./setup.py bdist_wheel -- -DCMAKE_VARIABLE=Values -- -Makefile_Flag
```
... but it can be a Visual Studio like flag, so the right thing to say here should be:
```
qiskit-aer$ python ./setup.py bdist_wheel -- -DCMAKE_VARIABLE=Values -- [-Makefile_Flag | /VisualStudio_Flag]
```
... but all in the same line.

Reply:
@atilag This section about scikit-build in our doc seems to contradict [scikit-build doc](https://scikit-build.readthedocs.io/en/latest/usage.html#command-line-options).

According to scikit-build doc, the way to pass variables is actually 

```
qiskit-aer$ python ./setup.py bdist_wheel [skbuild_opts] \
[-- cmake_opts] [-- build_tool_opts]
```

I don't understand where this part of our syntax example comes from: -DCMAKE_VARIABLE=Values. A Google search for DCMAKE_VARIABLE only returns a hit on this contributing file!

The example in our doc near line 103:

`qiskit-aer$ python ./setup.py bdist_wheel -j8`

seems to be explained incorrectly.

According to the scikit-build doc, the -j8 flag is actually being passed as skbuild_opts, although our doc says -j8 is being passed as a bulid_tool_opts. 

Our doc mentions STATIC_LINKING as a cmake_opt, but the example command doesn't set STATIC_LINKING explicitly.

So, overall, this section about scikit-build is very confusing to me. Could you help clarify what's trying to be said?

Reply:
For the example using -j8, the text says this is being passed to the underlying build system, so shouldn't the flag be entered as '-- -- -j8'? I'm basing this on [scikit-build docs](https://scikit-build.readthedocs.io/en/latest/usage.html#usage-setuptools-options).

Reply:
The format is the following:
```
python ./setup.py <action> -- parameters passed to CMake -- parameters passed to underlaying build system
```
The `-j8` is a parameter for Makefile (the underlaying build system in Mac and Linux), so for example, if I want to pass no parameters to CMake but pass some to Makefile, the command would be:
```
python ./setup.py bdist_wheel -- -- -j8
```
But if I do want to pass something to CMake:
```
python ./setup.py bdist_wheel -- -DPARAMTER_TO_CMAKE=Value -- -j8
```
So the `--` is like the delimiter